### DISCUSSION TITLE
Обрабатывать входящий Email

### DISCUSSION THREAD

**Discussion by brahmanda** [totumonline/totum-ru-issues-and-discussions#581] [Как это сделать?] [answered]
Добрый день !
Задача - обрабатывать входящий Email.
Для чего это нужно:
1) Поставщики отправляют счета на специальную почту, после обработки которой счета автоматом попадают в Totum.
2) Клиенты пишут в техподдержку на почту support@ и так же запросы попадают в Totum. Должна быть возможность ответить по Email.

**Comment by erpic-ru** [comment]
n8n вам в помощь: 

https://ru.docs.totum.online/n8nslack

https://n8n.io/workflows/154-listen-on-new-emails-on-a-imap-mailbox/

**Reply by ttm-alexeypolunin** [reply]
Наверное самый простой способ - это сторонний сервис типа мэйлгана, который может входящий imap преобразовывать в API-запрос (вебхук).

Если есть сисадмин в доступе, то мне тут чат говорит, что можно через fetchmail и munpack, но там я думаю два-три дня надо повозится с экспериментами.

Отправляет в Тотум функция emailSend в том числе с вложениями. Рекомендуется иметь подключенный внешний SMTP. Начиная с V5 тотум по умолчанию будет поставляться без Exim (с расчетом на внешний SMTP) тк сейчас без настройки почтового домена вообще нереально попасть во "входящие".

**Comment by brahmanda** [comment]
Спасибо за ответы! В интернете пишут, что fetchmail имеет проблемы с кириллицей. Сервисы n8n.io и mailgun с Российских IP блочат регистрацию, да и не хочется сторонним сервисом пользоваться.  Я вот нашел  на PHP  решение [php-imap](https://github.com/Webklex/php-imap). Если получить данные (тема, тело, файлы) средствами PHP, то как их передать в Totum корректно ?

**Reply by ttm-alexeypolunin** [reply]
Смотрите remotes: https://ru.docs.totum.online/api#part-4

Делаете внешний коннектор и на него скидываете данные.

**Reply by LostAlly** [reply]
>  Сервисы n8n.io и mailgun с Российских IP блочат регистрацию, да и не хочется сторонним сервисом пользоваться. 

n8n можно же на свой сервер поставить.

**Reply by brahmanda** [reply]
Отлично! Спасибо за советы.

**Comment by ttm-alexeypolunin** [comment, marked as answer]
Вот инструкция, все делается за 10 минут (ну после того как я 2 дня это поковырял)

**https://ru.docs.totum.online/imap_to_totum**

**Reply by ttm-alexeypolunin** [reply]
Доработал, теперь вообще огонь 🔥
